lib/dal/src/layer_db_types/content_types.rs

```diff
@@ -564,6 +568,77 @@ pub struct PropContentV1 {
     pub validation_format: Option<String>,
 }
 
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
+pub struct PropContentV2 {
+    pub timestamp: Timestamp,
+    /// The name of the [`Prop`].
+    pub name: String,
+    /// The kind of the [`Prop`].
+    pub kind: PropKind,
+    /// The kind of "widget" that should be used for this [`Prop`].
+    pub widget_kind: WidgetKind,
+    /// The configuration of the "widget".
+    pub widget_options: Option<WidgetOptions>,
+    /// A link to external documentation for working with this specific [`Prop`].
+    pub doc_link: Option<String>,
+    /// Embedded documentation for working with this specific [`Prop`].
+    pub documentation: Option<String>,
+    /// A toggle for whether or not the [`Prop`] should be visually hidden.
+    pub hidden: bool,
+    /// Props can be connected to eachother to signify that they should contain the same value
+    /// This is useful for diffing the resource with the domain, to suggest actions if the real world changes
+    pub refers_to_prop_id: Option<PropId>,
+    /// Connected props may need a custom diff function
+    pub diff_func_id: Option<FuncId>,
+    /// A serialized validation format JSON object for the prop.
+    pub validation_format: Option<String>,
+    /// Optional UI data that isn't needed in the graph
+    /// (Eventually, many other fields in content will be moved here, e.g. documentation/docLink)
+    pub ui_optionals: Option<HashMap<String, CasValue>>,
```

Comment:
@jkeiser 
Since we're introducing this I think it'd be good to at least have the comment be a bit more specific about when to add stuff here vs on widget options (even if it means that widget options is not to be used to any more things)

But I see that the secret kind for example is not moving (and I agree that it shouldn't, mostly because I feel widget options should be used for everything - even if I don't care about it strongly) and that raises some questions around when to use what

Reply:
@vbustamante our feeling here was that widget options should be more restricted because of how you specify it in asset builder: i.e. it seems like it's options for the corresponding WidgetKind--if you change the widget kind, the options are not necessarily valid anymore.

It seemed awkward to have `suggestSource()` be inside `WidgetDefinitionBuilder` rather than `PropBuilder`, basically. "If I'm not specifying a widget why do I have to have a WidgetDefinition?"

You could definitely slice it the other way though, and say "the widget really includes everything the UI displays," which actually means documentation/docLink should go there too ... it just doesn't quite fit the way we currently define it in the asset builder.